src/main/c/openssl/extconf.rb

```diff
@@ -30,11 +30,10 @@
 
 if ENV['OPENSSL_PREFIX']
   $CFLAGS += " -I #{ENV['OPENSSL_PREFIX']}/include"
-  $LIBS += " -l #{ENV['OPENSSL_PREFIX']}/lib/libssl.#{RbConfig::CONFIG['NATIVE_DLEXT']}"
-else
-  $LIBS += " -l libssl.#{RbConfig::CONFIG['NATIVE_DLEXT']}"
 end
 
+$LIBS += " -l libssl.#{RbConfig::CONFIG['NATIVE_DLEXT']}"
```

Comment:
It's also easy to make the mistake to pick up the system header but use another library with -Xcexts.remap